pg_search/src/postgres/customscan/pdbscan/exec_methods.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use crate::index::score::SearchIndexScore;
+use crate::index::SearchIndex;
+use crate::postgres::customscan::builders::custom_state::CustomScanStateWrapper;
+use crate::postgres::customscan::pdbscan::{make_tuple_table_slot, PdbScan};
+use pgrx::pg_sys::TupleTableSlot;
+use pgrx::{direct_function_call, pg_sys, IntoDatum};
+use tantivy::DocAddress;
+
+// TODO:  should these be GUCs?  I think yes, probably
```

Comment:
Are you thinking users should be able to specify them? If so, we might want to start a documentation page for perf optimization

Reply:
Yeah they might be good to expose via GUCs.